guide/docs/popular-topics/permissions.mdx

```diff
@@ -1,8 +1,382 @@
+
+assert only_mods_channel.permissions_for(kate).manage_messages
+```
+
+A slash command that checks whether the command author can add reactions in the current channel:
+
+```python
+@commands.slash_command(description="Checks whether you can add reactions!")
+async def can_add_reactions(inter: disnake.ApplicationCommandInteraction) -> None:
+    await inter.send("Sure you can!" if inter.permissions.add_reactions else "Sadly no :(")
+```
+
+## Querying permissions
+
+There are various ways to query permission information, and they also depend on the context you're
+in. This section tries to list all possible methods for getting permission info, grouped by context
+they can be used in.
+
+### Context-agnostic
+
+If you have an instance of any class implementing <DocsLink reference="disnake.abc.GuildChannel" />,
+then the easiest way to query permissions for a member/role is
+through <DocsLink reference="disnake.abc.GuildChannel.permissions_for" />.
+If you need a bit more lower level control, you can also get raw overwrite objects
+using <DocsLink reference="disnake.abc.GuildChannel.overwrites_for" />
+or <DocsLink reference="disnake.abc.GuildChannel.overwrites" />.
+
+### In prefix commands
+
+There are no direct methods on <DocsLink reference="disnake.ext.commands.Context" /> for querying permission
+information, however, you can still access <DocsLink reference="disnake.ext.commands.Context.channel" /> and
+call above-mentioned methods on it, though be sure to check that it's not <DocsLink reference="disnake.PartialMessageable" />
+as it doesn't have any permissions-related methods on it either.
+
+### In interactions
+
+When working with interactions (i.e., with application commands or components), there are two main methods
+defined on <DocsLink reference="disnake.Interaction" />: <DocsLink reference="disnake.Interaction.permissions" />
+(for getting interaction author's permissions) and <DocsLink reference="disnake.Interaction.app_permissions" />
+(for getting bot's permissions in interaction's channel).
+
+## How permissions are calculated
+
+Permissions can be a tough thing to understand at first, but understanding how permissions are
+applied can help you <!-- feel the pain of us mere mortals re-implementing discord client's obscure behaviour -->
+better grasp this concept.
+
+In case we're calculating permissions for a member, the strategy is as follows:
+
+-   If member is the guild owner, all permissions are granted - no questions asked. Otherwise..
+-   First we apply the `@everyone` role's permissions.
+-   Then, for each role the member has we apply it's permissions too.
+-   -   If at this point the member has the "Administrator" permission, all others are granted automatically
+        (the member is a guild-wide administrator, and, as such, is functionally equivalent to the owner except
+        they can't delete the guild).
+-   After that, if we're in a channel, the following is additionally done.
```

Comment:
Doesn't this always happen in a channel? I might be missing something here, though.

Reply:
While disnake does not provide any abstractions for that, one can in theory calculate permissions on a guild-level (for whatever reason). Since this section talks about permission calculation as a whole and not about concrete means to do it in disnake, i personally think this has a place to be.

Reply:
Alright, fair enough!